### DISCUSSION TITLE
请问写halo2 gate时候，Rotation如果超出（-1,0,1）的范围，会有什么不好的影响？

### DISCUSSION THREAD

**Discussion by mdj33** [z2o-k7e/halo2-step-by-step#16] [Q&A]
比如Rotation(-2)
meta.query_advice(advice[0], Rotation(-2));

我理解会计算w^-2, 如果不同Rotation多的话，就需要多计算几个不同的w^-x, 这样对性能有什么影响吗

**Comment by Demian101** [comment]
已由 @Demian101 @zhouyang 于第一次 OH 解答

大概就是行数变多，导致 proof size 变大
